resolver_srv.go

```diff
@@ -0,0 +1,43 @@
+package resolver
+
+import (
+	"context"
+	"net"
+
+	"github.com/miekg/dns"
+)
+
+// LookupSRV returns the DNS SRV for _service._proto.domain
+func (r LookupResolver) LookupSRV(ctx context.Context,
+	service, proto, name string) (string, []*net.SRV, error) {
+	target := ""
+	var netsrv []*net.SRV
+
+	if service == "" && proto == "" {
+		target = name
+	} else {
+		target = "_" + service + "._" + proto + "." + name
+	}
+	target = dns.Fqdn(target)
+
+	msg, err := r.h.Lookup(ctx, dns.CanonicalName(target), dns.TypeSRV)
+	if err != nil {
+		return "", nil, err
+	}
+	ForEachAnswer(msg, func(rr *dns.SRV) {
+		netsrv = append(netsrv, makeNetSRV(rr))
```

Comment:
this could lead to appending nil entries

Reply:
not anymore